### PR TITLE
`Once` semantics for `setup_{web|native}_logging`

### LINKED ISSUE #4893: `setup_native_logging` needs to handle its own `Once` semantics
Everything that calls into `setup_native_logging` (which is _a lot_) has to take care of wrapping it in `Once`-like semantics.

Let's implement these semantics in `setup_native_logging` directly.

### PR DESCRIPTION
**Pull request by teh-cmc**
- Fixes #4893 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4896/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4896/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4896/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4896)
- [Docs preview](https://rerun.io/preview/882cdd28a203faf740c68262522af73d16970804/docs) 
- [Examples preview](https://rerun.io/preview/882cdd28a203faf740c68262522af73d16970804/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_log/src/setup.rs [hunk 1]
```diff
@@ -3,57 +3,61 @@
 /// Directs [`log`] calls to stderr.
 #[cfg(not(target_arch = "wasm32"))]
 pub fn setup_native_logging() {
```
Comment (Wumpf) [inline crates/re_log/src/setup.rs:5]: why do we even distinguish native logging and web logging methods. Could just be one 🤔

**Review by Wumpf — APPROVED** [review]
nice, much better! 👍

File: crates/re_log/src/setup.rs (hunk 1 above)
Comment (teh-cmc) [reply crates/re_log/src/setup.rs:5]: that... is a good point :sweat_smile: lemme see

Comment (teh-cmc) [reply crates/re_log/src/setup.rs]: Yup, looks good; good call :+1:

**Pull request merged**